src/hooks/useArticlePagination.js

```diff
@@ -5,14 +5,18 @@ const DEVICE_WIDTH = 240
 
 export const useArticlePagination = (
   elementRef,
-  numOfSection
+  article,
+  subTitle,
+  showToc
 ) => {
-  const [currentSection, setCurrentSection] = useState(0)
+  const [currentSection, setCurrentSection] = useState(findLocatedSection(article.toc, subTitle))
   const [isLastPage, setIsLastPage] = useState(0)
   const prop = 'scrollLeft'
+  const numOfSection = article.sections.length
 
   useKeys({
     ArrowDown: () => {
+      if (showToc) return
```

Comment:
I think this line represents the main issue I see with this structure. We need to start thinking about better systems
* popups: so that their logic is not completely embedded in the components
* softkeys: to allow popups and other components to register (on show) and unregister (on hide) so we don't have to keep track of all the states in the handlers.